geode-core/src/test/java/org/apache/geode/security/IntegratedSecurityPeerAuthDUnitTest.java

```diff
@@ -0,0 +1,127 @@
+
+  private static MemberVM locator;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    Properties properties = new Properties();
+    properties.put(SECURITY_MANAGER, MySecurityManager.class.getName());
+    locator = cluster.startLocatorVM(0, properties);
+  }
+
+  @Test
+  public void startServer1WithPeerAuthInit_success() throws IOException {
+    Properties props = new Properties();
+    props.setProperty(SECURITY_PEER_AUTH_INIT, MyAuthInit.class.getName());
+    props.setProperty("security-name", "server-1");
+    cluster.startServerVM(1, props, locator.getPort());
+  }
+
+  @Test
+  public void startServer2_not_authorized() {
+    Properties props = new Properties();
+    props.setProperty(SECURITY_PEER_AUTH_INIT, MyAuthInit.class.getName());
+    props.setProperty("security-name", "server-2");
+    int locatorPort = locator.getPort();
+    cluster.getVM(2).invoke(() -> {
+      ServerStarterRule server = new ServerStarterRule();
+      server.withProperties(props).withConnectionToLocator(locatorPort).withAutoStart();
+      assertThatThrownBy(() -> server.before()).isInstanceOf(GemFireSecurityException.class)
+          .hasMessageContaining("server-2 not authorized for CLUSTER:MANAGE");
+    });
+  }
+
+  @Test
+  public void startServer3_not_authenticated() {
+    Properties props = new Properties();
+    props.setProperty(SECURITY_PEER_AUTH_INIT, MyAuthInit.class.getName());
+    props.setProperty("security-name", "server-3");
+    int locatorPort = locator.getPort();
+    cluster.getVM(3).invoke(() -> {
+      ServerStarterRule server = new ServerStarterRule();
+      server.withProperties(props).withConnectionToLocator(locatorPort).withAutoStart();
+      assertThatThrownBy(() -> server.before()).isInstanceOf(GemFireSecurityException.class)
+          .hasMessageContaining("Authentication error");
+    });
+  }
+
+  public static class MyAuthInit implements AuthInitialize {
+    @Override
+    public Properties getCredentials(Properties securityProps, DistributedMember server,
+        boolean isPeer) throws AuthenticationFailedException {
+      Properties properties = new Properties();
+      properties.setProperty("name", securityProps.getProperty("security-name"));
+      return properties;
+    }
+  }
+
+  public static class MySecurityManager implements SecurityManager {
+    @Override
+    public Object authenticate(Properties credentials) throws AuthenticationFailedException {
+      // server1 and server2 are authenticated, server3 is not
+      String name = credentials.getProperty("name");
+      if ("server-3".equals(name)) {
+        throw new AuthenticationFailedException("server-3 is not authenticated");
+      }
+      return name;
+    }
+
+    @Override
+    public boolean authorize(Object principal, ResourcePermission permission) {
+      // only server-1 and server-2 are authenticated, but only server-1 is authorized
+      if ("server-1".equals(principal)) {
```

Comment:
Does it still read as well when this is simplified to `return "server-1".equals(principal);`

Reply:
yup.